js/test/ValidateSPV.test.js

```diff
@@ -159,4 +171,89 @@ describe('ValidateSPV', () => {
       }
     });
   });
+
+  describe('#validateHeader', () => {
+    it('returns true if all elements of the bitcoin header are valid', () => {
+      for (let i = 0; i < validProof.length; i += 1) {
+        const { raw, hash, hash_le, height, merkle_root, merkle_root_le, prevhash } = validProof[i].confirming_header
+        const res = ValidateSPV.validateHeader(
+          raw, hash, hash_le, height, merkle_root, merkle_root_le, prevhash
+        );
+        assert.strictEqual(res, true);
+      }
+    });
+
+    it('throws error if any element of a header is invalid', () => {
+      for (let i = 0; i < badHeaders.length; i += 1) {
+        const { raw, hash, hash_le, height, merkle_root, merkle_root_le, prevhash } = badHeaders[i].header
+        try {
+          ValidateSPV.validateHeader(
+            raw, hash, hash_le, height, merkle_root, merkle_root_le, prevhash
+          )
+          assert(false, 'expected an error');
+        } catch (e) {
+          assert.include(e.message, badHeaders[i].e);
+        }
+      }
+    });
+  });
+
+  describe('#validateProof', () => {
+    it('returns true if all elements of the SPV Proof are valid', () => {
```

Comment:
`returns true if the proof object is syntactically valid`

Reply:
Fixed